libr/asm/asm.c

```diff
@@ -834,13 +845,12 @@ R_API RAsmCode* r_asm_massemble(RAsm *a, const char *buf) {
 				if (is_a_label) {
 					//if (stage != 2) {
 					if (ptr_start[1] != 0 && ptr_start[1] != ' ') {
-						char food[64];
 						*ptr = 0;
 						if (acode->code_align) {
 							off += (acode->code_align - (off % acode->code_align));
 						}
+						char food[64];
 						snprintf (food, sizeof (food), "0x%"PFMT64x"", off);
```

Comment:
Don't forget about `\0` at the end.

Reply:
the buffer is big enough to fit all ut64 representations in hex